lib/model/CSearchKey.cc

```diff
@@ -60,20 +59,17 @@ CSearchKey::CSearchKey(int detectorIndex,
                        function_t::EFunction function,
                        bool useNull,
                        model_t::EExcludeFrequent excludeFrequent,
-                       std::string fieldName,
-                       std::string byFieldName,
-                       std::string overFieldName,
-                       std::string partitionFieldName,
+                       const std::string& fieldName,
+                       const std::string& byFieldName,
+                       const std::string& overFieldName,
+                       const std::string& partitionFieldName,
```

Comment:
My inclination would be to pass by value and move into place, this is more idiomatic, and allows us to potentially avoid a copy, although probably not given how we use this object.

Reply:
I agree